lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -236,80 +241,70 @@ public JourneyResponse getJourneyResponse(CriCallbackRequest callbackRequest)
+        // Retrieve, store and check cri credentials
+        var accessToken = criApiService.fetchAccessToken(callbackRequest, criOAuthSessionItem);
+        var vcResponse =
+                criApiService.fetchVerifiableCredential(
+                        accessToken, callbackRequest, criOAuthSessionItem);
 
-            if (VerifiableCredentialStatus.PENDING.equals(vcResponse.getCredentialStatus())) {
-                criCheckingService.validatePendingVcResponse(vcResponse, clientOAuthSessionItem);
-                criStoringService.storeCriResponse(callbackRequest, clientOAuthSessionItem);
-            } else {
-                for (SignedJWT vc : vcResponse.getVerifiableCredentials()) {
-                    if (criOAuthSessionItem == null) {
-                        // We should never get here due to earlier null checks.
-                        // This is to satisfy a compile time warning
-                        throw new InvalidCriCallbackRequestException(
-                                ErrorResponse.INVALID_OAUTH_STATE);
-                    } else {
-                        verifiableCredentialJwtValidator.validate(
-                                vc,
-                                configService.getCriConfig(criOAuthSessionItem),
-                                clientOAuthSessionItem.getUserId());
-                    }
+        if (VerifiableCredentialStatus.PENDING.equals(vcResponse.getCredentialStatus())) {
+            criCheckingService.validatePendingVcResponse(vcResponse, clientOAuthSessionItem);
+            criStoringService.storeCriResponse(callbackRequest, clientOAuthSessionItem);
+        } else {
+            for (SignedJWT vc : vcResponse.getVerifiableCredentials()) {
+                if (criOAuthSessionItem == null) {
+                    // We should never get here due to earlier null checks.
+                    // This is to satisfy a compile time warning
+                    throw new InvalidCriCallbackRequestException(ErrorResponse.INVALID_OAUTH_STATE);
+                } else {
+                    verifiableCredentialJwtValidator.validate(
+                            vc,
+                            configService.getCriConfig(criOAuthSessionItem),
+                            clientOAuthSessionItem.getUserId());
                 }
-                criStoringService.storeVcs(
-                        callbackRequest.getCredentialIssuerId(),
-                        callbackRequest.getIpAddress(),
-                        callbackRequest.getIpvSessionId(),
-                        vcResponse.getVerifiableCredentials(),
-                        clientOAuthSessionItem);
-            }
-
-            return criCheckingService.checkVcResponse(
-                    vcResponse, callbackRequest, clientOAuthSessionItem, ipvSessionItem);
-        } finally {
-            if (ipvSessionItem != null) {
-                ipvSessionService.updateIpvSession(ipvSessionItem);
```

Comment:
Basically just removing this, since we no longer set any transient flags on the session 🎉